internal/controller/ols_app_server_reconciliator.go

```diff
@@ -363,10 +365,20 @@ func (r *OLSConfigReconciler) reconcilePrometheusRule(ctx context.Context, cr *o
 
 func (r *OLSConfigReconciler) reconcileTLSSecret(ctx context.Context, cr *olsv1alpha1.OLSConfig) error {
 	foundSecret := &corev1.Secret{}
-	secretValues, err := getSecretContent(r.Client, OLSCertsSecretName, r.Options.Namespace, []string{"tls.key", "tls.crt"}, foundSecret)
+	var err, lastErr error
+	var secretValues map[string]string
+	err = wait.PollUntilContextTimeout(ctx, 1*time.Second, ResourceCreationTimeout, true, func(ctx context.Context) (bool, error) {
+		secretValues, err = getSecretContent(r.Client, OLSCertsSecretName, r.Options.Namespace, []string{"tls.key", "tls.crt"}, foundSecret)
+		if err != nil {
+			lastErr = fmt.Errorf("secret: %s does not have expected tls.key or tls.crt. error: %w", OLSCertsSecretName, err)
+			return false, nil
```

Comment:
@raptorsun I see we never return the lastErr , we should be returning created error right ?

Reply:
If we return error in the wait loop, it will immediate abort the wait loop instead or trying once again. So we save the error but not return the error immediately. When timeout, we return the underlying error at line 379